integration/backup_storage_test.go

```diff
@@ -0,0 +1,136 @@
+package integration_test
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"net/http"
+
+	"github.com/AlekSi/pointer"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"oras.land/oras-go/pkg/context"
+
+	"github.com/percona/everest/client"
+)
+
+var _ = Describe("Backup Storages", func() {
+	Context("of type s3", func() {
+		createdBackupStorageName := ""
+		It("can be created", func() {
+			payload := client.CreateBackupStorageJSONRequestBody{
+				Type:        "s3",
+				Name:        "backup-storage-" + newTestSuffix(),
+				Url:         pointer.To("http://custom-url"),
+				Description: pointer.To("Dev storage"),
+				BucketName:  "percona-test-backup-storage",
+				Region:      "us-east-2",
+				AccessKey:   "sdfs",
+				SecretKey:   "sdfsdfsd",
+			}
+
+			resp, err := everestClient.CreateBackupStorage(context.Background(), testNs, payload)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(resp.StatusCode).To(Equal(http.StatusCreated))
+
+			body, err := io.ReadAll(resp.Body)
+			Expect(err).NotTo(HaveOccurred())
+			defer resp.Body.Close()
+
+			var parsed *client.BackupStorage
+			Expect(json.Unmarshal(body, &parsed)).To(Succeed())
```

Comment:
the generated client already provides method `CreateBackupStorageWithResponse` that returns parsed response and you can access it via `resp.JSON200`

Reply:
ah, missed that. Yeah that's more convenient to use..